lms/static/scripts/frontend_apps/index.js

```diff
@@ -6,21 +6,33 @@ import { createElement, render } from 'preact';
 
 import { Config } from './config';
 import BasicLtiLaunchApp from './components/BasicLtiLaunchApp';
+import CanvasAuthErrorDialog from './components/CanvasAuthErrorDialog';
 import FilePickerApp from './components/FilePickerApp';
 import { startRpcServer } from '../postmessage_json_rpc/server';
 
-// Create an RPC Server and start listening to postMessage calls.
-const rpcServer = startRpcServer();
-
 const rootEl = document.querySelector('#app');
 const config = JSON.parse(document.querySelector('.js-config').textContent);
 
+let rpcServer;
+if (config.mode === 'basic-lti-launch') {
+  // Create an RPC Server and start listening to postMessage calls.
+  rpcServer = startRpcServer();
+}
+
 render(
   <Config.Provider value={config}>
     {config.mode === 'basic-lti-launch' && (
       <BasicLtiLaunchApp rpcServer={rpcServer} />
     )}
     {config.mode === 'content-item-selection' && <FilePickerApp />}
+    {config.mode === 'canvas-auth-error-dialog' && (
+      <CanvasAuthErrorDialog
```

Comment:
Question here: I'm a bit stumped when to rely on the provider or directly pass the props in via config. Won't these all be available via the provider already?

Reply:
You're right - I hadn't thought too deeply about this when I put this draft together. I think it would be better to get it from the config where possible for consistency.